README.md

```diff
@@ -74,6 +74,65 @@ producer.deliver_messages
 
 Read the docs for [Kafka::Producer](http://www.rubydoc.info/gems/ruby-kafka/Kafka/Producer) for more details.
 
+### Partitioning
+
+Kafka topics are partitioned, with messages being assigned to a partition by the client. This allows a great deal of flexibility for the users. This section describes several strategies for partitioning and how they impact performance, data locality, etc.
+
+
+#### Load Balanced Partitioning
+
+When optimizing for efficiency, we either distribute messages as evenly as possible to all partitions, or make sure each producer always writes to a single partition. The former ensures an even load for downstream consumers; the latter ensures the highest producer performance, since message batching is done per partition.
+
+If no explicit partition is specified, the producer will look to the partition key or the message key for a value that can be used to deterministically assign the message to a partition. If there is a big number of different keys, the resulting distribution will be pretty even. If no keys are passed, the producer will randomly assign a partition. Random partitioning can be achieved even if you use message keys by passing a random partition key, e.g. `partition_key: rand(100)`.
+
+If you wish to have the producer write all messages to a single partition, simply generate a random value and re-use that as the partition key:
+
+```ruby
+partition_key = rand(100)
+
+producer.produce(msg1, topic: "messages", partition_key: partition_key)
+producer.produce(msg2, topic: "messages", partition_key: partition_key)
+
+# ...
+```
+
+You can also base the partition key on some property of the producer, for example the host name.
+
+#### Semantic Partitioning
+
```

Comment:
Nice stuff!

Is it worth being explicit that message order is only guaranteed within a partition?  This is implied here, but might be good to be explicit.